src/mediapipe_calculators/python_executor_calculator.cc

```diff
@@ -37,37 +37,53 @@ namespace mediapipe {
 const std::string INPUT_SIDE_PACKET_TAG = "PYTHON_NODE_RESOURCES";
 
 class PythonExecutorCalculator : public CalculatorBase {
-    /* 
-    TODO: Streaming support:
-        - timestamping
-        - loopback input
-    */
     std::shared_ptr<PythonNodeResource> nodeResources;
     std::unique_ptr<PyObjectWrapper<py::iterator>> pyIteratorPtr;
+    // The calculator manages timestamp for outputs to work independently of inputs
+    // this way we can support timestamp continuity for more than one request in streaming scenario. 
+    mediapipe::Timestamp outputsTimestamp;
 
-    void pushOutputs(CalculatorContext* cc, py::list pyOutputs) {
+    void pushOutputs(CalculatorContext* cc, py::list pyOutputs, mediapipe::Timestamp& timestamp, bool pushLoopback) {
         py::gil_scoped_acquire acquire;
         for (py::handle pyOutputHandle : pyOutputs) {
             py::object pyOutput = pyOutputHandle.cast<py::object>();
             nodeResources->pythonBackend->validateOvmsPyTensor(pyOutput);
             std::string outputName = pyOutput.attr("name").cast<std::string>();
             if (cc->Outputs().HasTag(outputName)) {
                 std::unique_ptr<PyObjectWrapper<py::object>> outputPtr = std::make_unique<PyObjectWrapper<py::object>>(pyOutput);
-                cc->Outputs().Tag(outputName).Add(outputPtr.release(), cc->InputTimestamp());
+                cc->Outputs().Tag(outputName).Add(outputPtr.release(), timestamp);
             }
         }
+        if (pushLoopback) {
+            cc->Outputs().Index(0).Add(std::make_unique<bool>(true).release(), timestamp);
+        }
+
+        timestamp++;
     }
 
 public:
     static absl::Status GetContract(CalculatorContract* cc) {
         LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->GetNodeName() << "] GetContract start";
         RET_CHECK(!cc->Inputs().GetTags().empty());
         RET_CHECK(!cc->Outputs().GetTags().empty());
-        for (auto& input : cc->Inputs()) {
-            input.Set<PyObjectWrapper<py::object>>();
+
+        // Input and output without a tag is treaded as loopback signal.
+        // Calculator expects one pair of such input and output indicating that there's a cycle
+        // All other inputs and outputs should come with a tag.
+        for (const std::string& tag : cc->Inputs().GetTags()) {
+            if (tag == "") {
+                cc->Inputs().Tag(tag).Set<bool>();
+            } else {
+                cc->Inputs().Tag(tag).Set<PyObjectWrapper<py::object>>();
+            }
         }
-        for (auto& output : cc->Outputs()) {
-            output.Set<PyObjectWrapper<py::object>>();
+
+        for (const std::string& tag : cc->Outputs().GetTags()) {
+            if (tag == "") {
+                cc->Outputs().Tag(tag).Set<bool>();
```

Comment:
Either way calculator will expect only one such input and output and any other would be ignored. 
This will be in documentation. Multiple untagged inputs and outputs is not expected configuration.